src/python/zquantum/core/interfaces/estimator.py

```diff
@@ -11,14 +12,17 @@ class Estimator(ABC):
     """
 
     def __init__(self):
-        pass
+        self.logger = logging.getLogger("z-quantum-core")
 
     @abstractmethod
     def get_estimated_expectation_values(
         self,
         backend: QuantumBackend,
         circuit: Circuit,
         target_operator: SymbolicOperator,
+        n_samples: int = None,
```

Comment:
I wonder if we couldn't put that in `__init__` ? Not sure if it makes more sense to have it here or separated...
One thing I don't like is the fact, that now we have n_samples specified in two places – in the definition of backend and here. And it's not clearly indicated that n_samples passed here overrides the n_samples from backend.

I wonder what would be the best way to deal with that... Perhaps we shouldn't pass n_samples and just assume we use the value from backend. Then we have a single source of truth, but I think it might be a little bit cumbersome to change the backend object every time before we want to run get_estimated_expectation_values.
So perhaps it's better to do it this way, just explicitly say in the docs that the n_samples here overrides the one defined in backend and throw a warning if the two don't match?

WDYT @max-radin @simonwa7 ?

Reply:
`n_samples` is something that can change per estimation routine, which is why I made it a method parameter. 

I don't think `QuantumBackend` should have `n_samples` as an attribute. Whether I'm right or wrong, I have no idea. But, I also think this PR is getting too big so I made the estimator take `n_samples` in a way that causes no side effects.

In the future, I would like to see objects using the backend tell it how many times to run something (rather than there being a single attribute that is set on initialization). But, again, I'm not sure if that's the way to go and left it for a future PR.

Reply:
Makes sense 👍